src/analyses/dependence_graph.cpp

```diff
@@ -195,10 +195,9 @@ void dep_graph_domaint::transform(
   // propagate control dependencies across function calls
   if(from->is_function_call())
   {
-    goto_programt::const_targett next=from;
-    ++next;
+    const goto_programt::const_targett next = std::next(from);
 
-    if(next==to)
+    if(next->function == to->function && next == to)
```

Comment:
Suggest commenting that just comparing function is not sufficient because of recursive calls